test/integration/targets/win_psmodule/tasks/main.yml

```diff
@@ -1,28 +1,97 @@
+# This file is part of Ansible
+
 # test code for the win_psmodule module when using winrm connection
-# (c) 2017, Daniele Lazzari <lazzari@mailup.com>
+# Copyright: (c) 2018, Wojciech Sciesinski <wojciech[at]sciesinski[dot]net>
+# Copyright: (c) 2017, Daniele Lazzari <lazzari@mailup.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
-# This file is part of Ansible
-#
-# Ansible is free software: you can redistribute it and/or modify
-# it under the terms of the GNU General Public License as published by
-# the Free Software Foundation, either version 3 of the License, or
-# (at your option) any later version.
-#
-# Ansible is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-# GNU General Public License for more details.
-#
-# You should have received a copy of the GNU General Public License
-# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
-
-
-- name: get facts
-  setup:
-
-- name: Perform integration tests for Powershell 5+
-  when: ansible_powershell_version >= 5
+- name: get PowerShell version
+  win_shell: '$PSVersionTable.PSVersion.Major'
+  register: powershell_major_version
+
+- name: update PackageManagement and PowerShellGet when PowerShell < 5.0
+  when: powershell_major_version.stdout | int < 5
+  block:
+    - name: get the TEMP path
+      win_shell: '$env:TEMP'
```

Comment:
You should be using the `win_output_dir` variable that is set up by `ansible-test` instead of getting the temp path yourself. This can be automatically set up by adding the following to the test role's `meta/main.yml`;

```
dependencies:
- prepare_win_tests
```